File: experimentos/control/cd/Cajero/Operaciones/ConsultaSaldo/ConsultaSaldo.cs
using System;

namespace Solucion;

public class ConsultaSaldo : Operacion
{
    private readonly IMovimientos _movimientos;

    public ConsultaSaldo(
        IUsuarios usuarios,
        IMovimientos movimientos) : base(usuarios)
    {
        _movimientos = movimientos;
    }

    public ResultadoConsultaSaldo Ejecutar(SolicitudConsultaSaldo solicitud)
    {
        ArgumentNullException.ThrowIfNull(solicitud);

        ResultadoConsultaSaldo resultado;

        if (UsuarioNoAutenticado(solicitud.Dni, solicitud.Clave, out var usuario, out var codigo, out var mensaje))
        {
            resultado = new ResultadoConsultaSaldo(false, null, codigo, mensaje);
        }
        else
        {
            resultado = new ResultadoConsultaSaldo(
                true, _movimientos.ObtenerSaldoActualCuenta(usuario), "CS00", null);
        }

        return resultado;
    }
}

File: solucion_modelo/operaciones/saldo/consulta_saldo.py
from operaciones.saldo.parametros_consulta_saldo import ParametrosConsultaSaldo
from tipos import *
from estado import Estado


def consulta_saldo(solicitud: ParametrosConsultaSaldo):
    estado = Estado()

    if solicitud.dni not in estado.usuarios.keys():
        return RESULTADO.UsuarioInexistente

    if estado.usuarios[solicitud.dni].clave != solicitud.clave:
        return RESULTADO.ClaveIncorrecta

    saldo = estado.usuarios[solicitud.dni].saldo

    print(f"El saldo del usuario DNI {solicitud.dni} es: {saldo}")

    return RESULTADO.OK
